kernel/combinatorics/hilb.cc

```diff
@@ -1400,5 +1426,742 @@ void hLookSeries(ideal S, intvec *modulweight, ideal Q, intvec *wdegree, ring ta
+  
+  poly w, wi;
+  ideal Jwi;
+
+  while(lpcnt < idorb.size())
+  {
+    w=NULL;
+    w=polist[lpcnt];
+
+    if(lpcnt >=1)
+    {
+      if(DegMon(idorb[lpcnt]->m[0])!=0) 
+      {
+        C.push_back(1);
+      }
+      else
+        C.push_back(0);
+    }
+    else
+      C.push_back(1);
+
+    ds=DegMon(w);
+    lpcnt++;
+
+    for(is=1;is<=lV;is++)
+    {
+      wi=NULL;
+      wi=pCopy(w);
+      p_SetExp(wi,(ds*lV)+is,1,r);
+      p_Setm(wi,r);
+      Jwi=NULL;
+      Jwi=idInit(1,1);
+
+      Jwi=colonIdeal(S,wi,lV,Jwi);
+      ps=(*POS)(Jwi,wi,idorb,polist,trInd);
+      
+      if(ps==0)
+      {
+        idorb.push_back(Jwi);
+        polist.push_back(wi);
+        row.push_back(1);
+      }
+      else
+      {
+        row[ps-1]=row[ps-1]+1;
+        idDelete(&Jwi);    
+      }
+    }
+    mat.push_back(row);
+    sz=row.size();
+    row.clear();
+    row.resize(sz,0);
+  }
+
+  for(is=idorb.size()-1;is>=0;is--)
+  {  
+    idDelete(&idorb[is]);
+  }
+  for(is=polist.size()-1;is>=0;is--)
+  {
+    pDelete(&polist[is]);
+  }
+  
+  idorb.resize(0); 
+  polist.resize(0);
+
+  row.resize(0);
+
+  int ir,ic;  
+#if 0
+  for(ir=0;ir<mat.size();ir++)
+  {
+    for(ic=0;ic<mat[ir].size();ic++)
+    {
+      printf("%d,",mat[ir][ic]);
+    }
+    printf("\n");
+  }
+  printf("rhs column matrix::\n");
+  for(ic=0;ic<C.size();ic++)
+  printf("%d,",C[ic]);
+  printf("\nlength of the Orbit==%ld\n", C.size());
+#endif
+
+  char** tt=(char**)omalloc(sizeof(char*));
+  tt[0]=omStrDup("t");
+  TransExtInfo p;
+  p.r = rDefault(0, 1, tt);
+  coeffs cf=nInitChar(n_transExt,&p);  
+  
+  char** xx=(char**)omalloc(sizeof(char*));
+  xx[0]= omStrDup("x");
+  ring R=rDefault(cf, 1, xx);  
+  rChangeCurrRing(R);  
+  /*
+   * matrix corresponding to the orbit of the ideal
+   */
+  int lO=C.size();//size of the orbit
+  matrix mR=mpNew(lO,lO);
+  matrix cMat=mpNew(lO,1);
+
+  for(ir=0;ir<lO;ir++)
+  {
+    for(ic=0;ic<mat[ir].size();ic++)
+    {
+      if(mat[ir][ic]!=0)
+      {
+        mR->m[lO*ir+ic]=p_ISet(mat[ir][ic],R);
+        p_SetCoeff(mR->m[lO*ir+ic], n_Mult(pGetCoeff(mR->m[lO*ir+ic]),n_Param(1,R->cf),R->cf),R);
+      }
+    }
+    if(C[ir]!=0)
+    {
+      cMat->m[ir]=p_ISet(C[ir],R);
+    }
+    mat[ir].resize(0);
+  }
+  mat.resize(0);
+  C.resize(0);
+  matrix u;
+  unitMatrix(lO,u); //unit matrix
+  matrix gMat=mp_Sub(u,mR,R);
+  matrix pMat;
+  matrix lMat;
+  matrix uMat;
+  luDecomp(gMat, pMat, lMat, uMat, R);
+  matrix H_serVec = mpNew(lO, 1);
+  matrix Hnot;
+  luSolveViaLUDecomp(pMat, lMat, uMat, cMat, H_serVec, Hnot);
+  
+  mp_Delete(&mR,R);
+  mp_Delete(&u,R);
+  mp_Delete(&pMat,R);
+  mp_Delete(&lMat,R);
+  mp_Delete(&uMat,R);
+  mp_Delete(&cMat,R);
+  mp_Delete(&gMat,R);
+  mp_Delete(&Hnot,R);
+  //print the Hilbert series and Orbit length
+  PrintLn();
+  pWrite(H_serVec->m[0]);
+  Print("\nOrbit size = %d\n", lO);
+
+  omFree(tt[0]);
+  omFree(tt);
+  omFree(xx[0]);
+  omFree(xx);
+  rChangeCurrRing(r);
+}
```

Comment:
in general: 
- make all helper routines static
- add comments about the task of each routine, it prerequisites and output (especially: what is copied or not)